### PR TITLE
Add qe coverage for Bugzilla 1649928

### PR DESCRIPTION
**Pull request by hkx303**
**Description**
Added automation test case for [Bugzilla 1649928](https://bugzilla.redhat.com/show_bug.cgi?id=1649928)

**Test Result**
```
(venv) [hkx303@kuhuang virtwho]$ pytest test_ui_virtwho.py::test_positive_overview_label_name
2019-10-30 08:29:51 - conftest - DEBUG - Registering custom pytest_configure

==================================== test session starts =====================================
platform linux -- Python 3.7.3, pytest-4.0.2, py-1.8.0, pluggy-0.12.0
shared_function enabled - OFF - scope:  - storage: file
rootdir: /home/hkx303/Documents/CI/robottelo/robottelo, inifile:
plugins: services-1.3.1, forked-1.0.2, mock-1.10.4, cov-2.7.1, xdist-1.29.0
collecting ... 2019-10-30 08:29:51 - conftest - DEBUG - BZ deselect is disabled in settings

collected 1 item                                                                             

test_ui_virtwho.py .                                                                   [100%]

=========================== 1 passed, 3 warnings in 290.33 seconds ===========================
```

### CODE REVIEW

File: tests/foreman/virtwho/test_ui_virtwho.py [hunk 1]
```diff
@@ -626,3 +626,68 @@ def test_positive_virtwho_manager_role(session, test_name, form_data):
...
+        fields['proxy_label'] = 'HTTP Proxy'
+        fields['no_proxy_label'] = 'Ignore Proxy'
+        for key, value in fields.items():
+            assert results['overview'][key] == value
+        session.virtwho_configure.edit(name, blacklist)
+        results = session.virtwho_configure.read(name)
+        del fields['filter_hosts_label']
+        if hypervisor_type == 'esx':
+            del fields['filter_host_parents_label']
+            fields['exclude_host_parents_label'] = 'Exclude Host Parents'
```
Comment (mirekdlugosz) [inline tests/foreman/virtwho/test_ui_virtwho.py]: You have set `exclude_host_parents_label` already in lin 678.

Comment (hkx303) [reply tests/foreman/virtwho/test_ui_virtwho.py]: nope, lin 678 is `filter_host_parents_label` , not this `exclude_host_parents_label`

File: tests/foreman/virtwho/test_ui_virtwho.py [hunk 2]
```diff
@@ -626,3 +626,68 @@ def test_positive_virtwho_manager_role(session, test_name, form_data):
...
+            'hypervisor_id_label': 'Hypervisor ID',
+            'filtering_label': 'Filtering',
+            'debug_label': 'Enable debugging output?'
+        }
+        for key, value in fields.items():
+            assert results['overview'][key] == value
+        session.virtwho_configure.edit(
+            name, dict({'proxy': http_proxy,
+                        'no_proxy': no_proxy},
+                       **whitelist))
```
[the remark this answers is not in the document]
Comment (hkx303) [reply tests/foreman/virtwho/test_ui_virtwho.py]: ok, thanks

File: tests/foreman/virtwho/test_ui_virtwho.py [hunk 3]
```diff
@@ -626,3 +626,68 @@ def test_positive_virtwho_manager_role(session, test_name, form_data):
...
+
+
+def test_positive_overview_label_name(form_data, session):
+    """Verify the label name on virt-who config Overview Page.
+
+    :id: 68422042-648c-4652-a52b-5bf42462d2ae
+
+    :CaseLevel: Integration
+
+    :CaseImportance: Medium
```
Comment (vijay8451) [inline tests/foreman/virtwho/test_ui_virtwho.py]: please add bugzilla as part of test docstring ..

```
:BZ: 1649928
```

Comment (hkx303) [reply tests/foreman/virtwho/test_ui_virtwho.py]: ok, thanks

**Comment by hkx303** [comment]
> Overall looks OK, but seeing that you read labels three times, I wonder if this could be optimized. Maybe work only on hypervisor type `esx` (as it seems to have some additional fields)? And maybe set proxies and whitelist right at the start?

yes, for type `esx`, we have some additional fields, I optimized it so that we can only read labels twice

**Comment by hkx303** [comment]
New test result:

```
(venv) [hkx303@kuhuang virtwho]$ pytest test_ui_virtwho.py::test_positive_overview_label_name
2019-10-30 11:26:29 - conftest - DEBUG - Registering custom pytest_configure

==================================== test session starts =====================================
platform linux -- Python 3.7.3, pytest-4.0.2, py-1.8.0, pluggy-0.12.0
shared_function enabled - OFF - scope:  - storage: file
rootdir: /home/hkx303/Documents/CI/robottelo/robottelo, inifile:
plugins: services-1.3.1, forked-1.0.2, mock-1.10.4, cov-2.7.1, xdist-1.29.0
collecting ... 2019-10-30 11:26:29 - conftest - DEBUG - BZ deselect is disabled in settings

collected 1 item                                                                             

test_ui_virtwho.py .                                                                   [100%]

====================================== warnings summary ======================================
tests/foreman/virtwho/test_ui_virtwho.py::test_positive_overview_label_name
  /home/hkx303/Documents/CI/robottelo/venv/lib/python3.7/site-packages/widgetastic/widget/base.py:815: DeprecationWarning: inspect.getargspec() is deprecated since Python 3.0, use inspect.signature() or inspect.getfullargspec()
    c_args, c_varargs, c_keywords, c_defaults = inspect.getargspec(condition)
  /home/hkx303/Documents/CI/robottelo/venv/lib/python3.7/site-packages/widgetastic/browser.py:721: DeprecationWarning: use driver.switch_to.alert instead
    return self.selenium.switch_to_alert()
  /home/hkx303/Documents/CI/robottelo/venv/lib/python3.7/site-packages/widgetastic/browser.py:721: DeprecationWarning: use driver.switch_to.alert instead
    return self.selenium.switch_to_alert()

-- Docs: https://docs.pytest.org/en/latest/warnings.html
=========================== 1 passed, 3 warnings in 137.23 seconds ===========================
```

File: tests/foreman/virtwho/test_ui_virtwho.py (hunk 1 above)
Comment (mirekdlugosz) [reply tests/foreman/virtwho/test_ui_virtwho.py]: Ah, you are right. Don't know what I was thinking.

File: tests/foreman/virtwho/test_ui_virtwho.py [hunk 4]
```diff
@@ -626,3 +626,67 @@ def test_positive_virtwho_manager_role(session, test_name, form_data):
...
+            'proxy_label': 'HTTP Proxy',
+            'no_proxy_label': 'Ignore Proxy'
+        }
+        if hypervisor_type == 'esx':
+            fields['filter_host_parents_label'] = 'Filter Host Parents'
+        for key, value in fields.items():
+            assert results['overview'][key] == value
+        results = session.virtwho_configure.read(name)
+        for key, value in fields.items():
+            assert results['overview'][key] == value
```
Comment (eko999) [inline tests/foreman/virtwho/test_ui_virtwho.py]: duplicated steps?

File: tests/foreman/virtwho/test_ui_virtwho.py [hunk 5]
```diff
@@ -626,3 +626,67 @@ def test_positive_virtwho_manager_role(session, test_name, form_data):
...
+    :BZ: 1649928
+
+    :CaseLevel: Integration
+
+    :CaseImportance: Medium
+    """
+    name = gen_string('alpha')
+    form_data['name'] = name
+    hypervisor_type = form_data['hypervisor_type']
+    form_data['proxy'] = 'test.rexample.com:3128'
```
Comment (eko999) [inline tests/foreman/virtwho/test_ui_virtwho.py]: a little typo, it's not rexample, it should be example:)

Comment (hkx303) [reply tests/foreman/virtwho/test_ui_virtwho.py]: oh, copy from test_positive_proxy_option, I will modify that as well

**Comment by hkx303** [comment]
Test Result for the last commit
```
(venv) [hkx303@kuhuang virtwho]$ pytest test_ui_virtwho.py::test_positive_overview_label_name
2019-10-31 08:52:42 - conftest - DEBUG - Registering custom pytest_configure

==================================== test session starts =====================================
platform linux -- Python 3.7.3, pytest-4.0.2, py-1.8.0, pluggy-0.12.0
shared_function enabled - OFF - scope:  - storage: file
rootdir: /home/hkx303/Documents/CI/robottelo/robottelo, inifile:
plugins: services-1.3.1, forked-1.0.2, mock-1.10.4, cov-2.7.1, xdist-1.29.0
collecting ... 2019-10-31 08:52:42 - conftest - DEBUG - BZ deselect is disabled in settings

collected 1 item                                                                             

test_ui_virtwho.py .                                                                   [100%]

====================================== warnings summary ======================================
tests/foreman/virtwho/test_ui_virtwho.py::test_positive_overview_label_name
  /home/hkx303/Documents/CI/robottelo/venv/lib/python3.7/site-packages/widgetastic/widget/base.py:815: DeprecationWarning: inspect.getargspec() is deprecated since Python 3.0, use inspect.signature() or inspect.getfullargspec()
    c_args, c_varargs, c_keywords, c_defaults = inspect.getargspec(condition)
  /home/hkx303/Documents/CI/robottelo/venv/lib/python3.7/site-packages/widgetastic/browser.py:721: DeprecationWarning: use driver.switch_to.alert instead
    return self.selenium.switch_to_alert()
  /home/hkx303/Documents/CI/robottelo/venv/lib/python3.7/site-packages/widgetastic/browser.py:721: DeprecationWarning: use driver.switch_to.alert instead
    return self.selenium.switch_to_alert()

-- Docs: https://docs.pytest.org/en/latest/warnings.html
=========================== 1 passed, 3 warnings in 118.87 seconds ===========================
```

**Pull request merged**